src/test/java/org/opensearch/neuralsearch/processor/RRFSearchIT.java

```diff
@@ -0,0 +1,171 @@
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.neuralsearch.util.TestUtils.DEFAULT_USER_AGENT;
+import static org.opensearch.neuralsearch.util.TestUtils.DELTA_FOR_SCORE_ASSERTION;
+
+public class RRFSearchIT extends BaseNeuralSearchIT {
+
+    private int currentDoc = 1;
+    private static final String RRF_INDEX_NAME = "rrf-index";
+    private static final String RRF_SEARCH_PIPELINE = "rrf-search-pipeline";
+
+    @SneakyThrows
+    public void testRRF() {
+        String modelId = prepareModel();
+        String ingestPipelineName = "rrf-ingest-pipeline";
+        createPipelineProcessor(modelId, ingestPipelineName, ProcessorType.TEXT_EMBEDDING);
+        Settings indexSettings = Settings.builder().put("index.knn", true).put("default_pipeline", ingestPipelineName).build();
+        String indexMappings = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("properties")
+            .startObject("id")
+            .field("type", "text")
+            .endObject()
+            .startObject("passage_embedding")
+            .field("type", "knn_vector")
+            .field("dimension", "768")
+            .startObject("method")
+            .field("engine", "lucene")
+            .field("space_type", "l2")
+            .field("name", "hnsw")
+            .endObject()
+            .endObject()
+            .startObject("text")
+            .field("type", "text")
+            .endObject()
+            .endObject()
+            .endObject()
+            .toString();
+        // Removes the {} around the string, since they are already included with createIndex
+        indexMappings = indexMappings.substring(1, indexMappings.length() - 1);
+        String indexName = "rrf-index";
+        createIndex(indexName, indexSettings, indexMappings, null);
+        addRRFDocuments();
+        createDefaultRRFSearchPipeline();
+
+        Map<String, Object> results = searchRRF(modelId);
+        Map<String, Object> hits = (Map<String, Object>) results.get("hits");
+        ArrayList<HashMap<String, Object>> hitsList = (ArrayList<HashMap<String, Object>>) hits.get("hits");
+        assertEquals(3, hitsList.size());
+        assertEquals(0.016393442, (Double) hitsList.getFirst().get("_score"), DELTA_FOR_SCORE_ASSERTION);
+        assertEquals(0.016129032, (Double) hitsList.get(1).get("_score"), DELTA_FOR_SCORE_ASSERTION);
+        assertEquals(0.015873017, (Double) hitsList.getLast().get("_score"), DELTA_FOR_SCORE_ASSERTION);
```

Comment:
you need to have a resource cleanup block, please check how it's done in other tests ([code ref](https://github.com/opensearch-project/neural-search/blob/main/src/test/java/org/opensearch/neuralsearch/processor/NormalizationProcessorIT.java#L117-L118))